### PR TITLE
test(primitives/types)

### LINKED ISSUE #216: Unit tests - M1 - types
Add  tests for primitive types https://github.com/LimeChain/gosemble/tree/develop/primitives/types

* [x] AccountData - @radkomih 
* [x] AccountInfo - @radkomih 
* [x] ApiItem - @failfmi 
* [x] ApplyExtrinsicResult - @radkomih 
* [x] ArithmeticError - @radkomih 
* [x] Authority - @radkomih 
* [x] BalanceStatus - @radkomih 
* [x] BlockLength - @radkomih 
* [x] BlockWeights - @radkomih 
* [x] Weight - @radkomih 
* [x] DispatchOutcome - @failfmi 
* [x] FeeDetails - @radkomih 
* [x] InclusionFee - @radkomih 
* [x] Callable - @radkomih 
* [x] CheckInherentsResult - @radkomih 
* [x] InherentError - @radkomih 
* [x] DigestItem - @radkomih 
* [x] Digest - @radkomih 
* [x] DispatchClass - @radkomih 
* [x] PerDispatchClass - @radkomih 
* [x] DispatchInfo - @radkomih 
* [x] ConsumedWeight - @radkomih 
* [x] DispatchResult - @radkomih
* [x] DispatchResultWithPostInfo - @radkomih 
* [x] Event - @radkomih  
* [x] Era - @radkomih 
* [x] ExtrinsicPhase - @failfmi
* [x] SignedExtra - @radkomih 
* [x] SignedPayload - @radkomih 
* [x] ExtrinsicSignature - @radkomih 
* [x] H256, H512, Blake2bHash - @nicksinch 
* [x] Header - @nicksinch 
* [x] Identity fee - @failfmi
* [x] InherentData - @failfmi 
* [x] Invalid transaction - @failfmi
* [x] Last runtime upgrade info - @failfmi
* [x] Lookup - @nicksinch 
* [ ] Metadata
* [x] Module - @nicksinch 
* [x] Multi-address - @radkomih 
* [x] MultiSignature - @failfmi
* [x] Pays - @failfmi
* [ ] PerThings
* [x] PostDispatchInfo - @failfmi
* [x] PublicKey - @failfmi
* [x] RawOrigin - @failfmi  
* [x] RuntimeDbWeight - @failfmi 
* [x] RuntimeDIspatchInfo - @failfmi 
* [x] RuntimeVersion - @failfmi 
* [x] SessionKey - @failfmi  
* [x] SignatureEcdsa - @failfmi  
* [x] SignatureEd25519 - @failfmi  
* [x] SignatureSr25519 - @failfmi  
* [x] TimestampError - @failfmi  
* [x] TokenError - @failfmi 
* [x] TransactionOutcome - @failfmi
* [x] TransactionSource - @failfmi
* [x] TransactionValidityError - @failfmi
* [x] TransactionValidityResult - @failfmi
* [x] TransactionalError - @failfmi
* [x] UnknownTransaction - @failfmi
* [x] ValidTransaction - @failfmi
* [x] VersionedAuthorityList - @failfmi
* [x] WeightPerClass - @failfmi

### PR DESCRIPTION
**Pull request by radkomih**
**Detailed description**:

* [x] test `ConsumedWeight` (fix `CheckedAccrue` method)
* [x] test `DispatchResult`
* [x] test `DispatchResultWithPostInfo`
* [x] test `Event`
* [x] finish `Era` tests
* [x] test `SignedExtra`

**Which issue(s) this PR fixes**:
Fixes #216 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by radkomih** [comment]
need to fix something

**Review by nicksinch — DISMISSED** [review]

File: primitives/types/dispatch_result_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	sc "github.com/LimeChain/goscale"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	postDispatchInfo2 = PostDispatchInfo{
+		ActualWeight: sc.NewOption[Weight](WeightFromParts(1, 2)),
+		PaysFee:      PaysYes,
+	}
```
Comment (failfmi) [inline primitives/types/dispatch_result_test.go]: You can use `postDispatchInfo`

File: primitives/types/dispatch_result_with_post_info_test.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+import (
+	"testing"
+
+	sc "github.com/LimeChain/goscale"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	dispatchInfo2 = DispatchInfo{
```
Comment (failfmi) [inline primitives/types/dispatch_result_with_post_info_test.go]: This can be renamed to something else

File: primitives/types/event_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+)
+
+func Test_NewEvent(t *testing.T) {
+	expectedEvent := sc.NewVaryingData(sc.U8(1), sc.U8(2), sc.FixedSequence[sc.U8]{3, 4}, sc.U8(5))
+
+	assert.Equal(t, expectedEvent, NewEvent(1, 2, sc.FixedSequence[sc.U8]{3, 4}, sc.U8(5)))
+}
+
+func Test_DecodeEvents(t *testing.T) {
+	eventRecordsBytes := []byte{}
```
Comment (failfmi) [inline primitives/types/event_test.go]: `var eventRecordsBytes []byte`

File: primitives/types/event_test.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	eventRecords = sc.Sequence[EventRecord]{eventRecord2, eventRecord2}
+)
+
+func Test_NewEvent(t *testing.T) {
+	expectedEvent := sc.NewVaryingData(sc.U8(1), sc.U8(2), sc.FixedSequence[sc.U8]{3, 4}, sc.U8(5))
+
+	assert.Equal(t, expectedEvent, NewEvent(1, 2, sc.FixedSequence[sc.U8]{3, 4}, sc.U8(5)))
+}
+
+func Test_DecodeEvents(t *testing.T) {
```
Comment (failfmi) [inline primitives/types/event_test.go]: `DecodeEvent` should take the event you would like to decode.

File: primitives/types/extra_test.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
+package types
+
+import (
```
Comment (failfmi) [inline primitives/types/extra_test.go:3]: Coverage is missing here

File: primitives/types/dispatch_result_test.go (hunk 1 above)
Comment (radkomih) [reply primitives/types/dispatch_result_test.go]: there is one already declared in `post_dispatch_info_test.go`
I will rename it to something else

Comment (failfmi) [reply primitives/types/dispatch_result_test.go]: It is the same

Comment (radkomih) [reply primitives/types/dispatch_result_test.go]: yes, but I prefer to use something that is declared in the same file

File: primitives/types/testextra.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
+package types
```
Comment (failfmi) [inline primitives/types/testextra.go]: Can we declare this in a test file, initialiser can be lowercase as well?

File: primitives/types/event.go [hunk 7]
```diff
@@ -11,22 +11,9 @@ type Event = sc.VaryingData
...
-	size := int(compactSize.ToBigInt().Int64())
-
-	sequence := make(sc.Sequence[EventRecord], size)
-	for i := 0; i < size; i++ {
-		sequence[i] = DecodeEventRecord(buffer)
-	}
-
-	return sequence
-}
-
```
Comment (failfmi) [inline primitives/types/event.go:29]: let's return this

Comment (failfmi) [reply primitives/types/event.go:29]: or at least have it commented

**Review by failfmi — APPROVED** [review]

**Pull request merged**